src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs

```diff
@@ -15,7 +15,7 @@ public static class SentryOptionsDiagnosticExtensions
     /// <param name="options">The Sentry options.</param>
     public static void AddDiagnosticSourceIntegration(this SentryOptions options)
     {
-        if (options.Integrations?.OfType<SentryDiagnosticListenerIntegration>().Any() is true)
+        if (options.Integrations.Any(integration => integration.Type == typeof(SentryDiagnosticListenerIntegration)))
```

Comment:
Now with the type, we prob should go with a dictionary Type:instance
We're doing allocations in each of the LINQ calls, and iterating the lists

Reply:
That was my previous modification but I've had issues in unit test where we use substitutes as they were sharing the same "RuntimeType"  type.

Also, it was not clear to me if we can or not have several times the same types.

At the end, I use `List<Tuple>` as I felt like it was more close to previous code.

Reply:
Fair, Tuple is the way to go unless we'd want to break that behavior.
Still dropping the LINQ usage is potential area of improvement if we're looking for more.

Reply:
Still, overall it feels a bit complex. How would you feel if those default integrations would be stored in an enum flag? During the config you would just need to remove the flags in the `DisableXXX` methods. It requires a bit more work though.

Reply:
```csharp
[Flags]
internal enum DefaultIntegrations
{
    AutoSessionTrackingIntegration = 1 << 0,
    AppDomainUnhandledExceptionIntegration = 1 << 1,
    AppDomainProcessExitIntegration = 1 << 2,
    UnobservedTaskExceptionIntegration = 1 << 3,
    NetFxInstallationsIntegration = 1 << 4,
    SentryDiagnosticListenerIntegration = 1 << 5,
    WinUiUnhandledExceptionIntegration = 1 << 6,
}
```

Reply:
I gave it a shot in my last commit.